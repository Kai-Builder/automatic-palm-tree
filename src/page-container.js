import {main, navigation, pageContainer} from './page-container.scss';
import {AnonymousRoute} from './anonymous-route';
import {HomeRoute} from './home-route';
import {Navigation} from './navigation';
import React from 'react';
import {RestrictedRouteA} from './restricted-route-a';
import {RestrictedRouteB} from './restricted-route-b';

export const PageContainer = () => (
  <section className={pageContainer}>
    <Navigation className={navigation}/>
    <HomeRoute className={main} />
    <AnonymousRoute className={main}/>
    <RestrictedRouteA className={main}/>
    <RestrictedRouteB className={main}/>
  </section>
);
