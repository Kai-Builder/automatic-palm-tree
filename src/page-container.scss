.pageContainer {
  display: grid;

  grid-template-areas:
    'navigation navigation'
    'main main';

  grid-template-columns: repeat(2, fr);
  grid-template-rows: repeat(2, fr);
 }

.navigation {
  grid-area: navigation;
}

.main {
  grid-area: main;
}
