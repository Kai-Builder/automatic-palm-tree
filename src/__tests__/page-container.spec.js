import {PageContainer} from '../page-container';
import React from 'react';
import {shallow} from 'enzyme';

describe('PageContainer', () => {
  const requiredProps = () => ({
  });
  const render = (props = requiredProps()) => shallow(<PageContainer {...props}/>);

  test('component renders', () => {
    const wrapper = render();

    expect(wrapper).toMatchSnapshot();
  });
});
