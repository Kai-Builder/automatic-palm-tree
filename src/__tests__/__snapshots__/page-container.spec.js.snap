// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageContainer component renders 1`] = `
<section
  className="pageContainer"
>
  <Navigation
    className="navigation"
  />
  <HomeRoute
    className="main"
  />
  <AnonymousRoute
    className="main"
  />
  <RestrictedRouteA
    className="main"
  />
  <RestrictedRouteB
    className="main"
  />
</section>
`;
